.ci/lint-cpp.sh

```diff
@@ -32,9 +34,12 @@ get_omp_pragmas_without_num_threads() {
         'pragma omp parallel' \
     | grep -v ' num_threads'
 }
+# https://www.gnu.org/software/grep/manual/html_node/Exit-Status.html
+set +e +o pipefail
```

Comment:
Could you add a brief description in prose why this is necessary? It took me a second to see that `get_omp_pragmas_without_num_threads` runs `grep`

Reply:
updated in https://github.com/microsoft/LightGBM/pull/6266/commits/e4fa87d2743ee91ccf8b89c43a54e212f6971183, let me know what you think

Reply:
Reads nicely, I think it should be "returns a non-0 exit code if **no** lines were found" though, right? 🤔

Reply:
ah!! yes you're totally right, sorry 😓

Reply:
fixed in https://github.com/microsoft/LightGBM/pull/6266/commits/9d24415f0423fb4ab4e245bfa0a6575d8af3ad82